oteapi/models/genericconfig.py

```diff
@@ -0,0 +1,60 @@
+"""Generic data model for configuration attributes."""
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+
+class AttrDict(BaseModel):
+    """An object whose attributes can also be accessed through
+    subscription, like with a dictionary."""
+
+    class Config:
+        """Class for configuration of pydantic models."""
+
+        extra = "allow"
+
+    def __contains__(self, name: Any) -> bool:
+        """Enable using the 'in' operator on this object."""
+        return self.__dict__.__contains__(name)
+
+    def __delitem__(self, key: str) -> None:
+        """Enable deletion access through subscription."""
+        if key in dir(self):
+            self.__delattr__(key)
+        else:
+            raise KeyError(key)
+
+    def __getitem__(self, key: str) -> Any:
+        """Enable read access through subscription."""
+        if key in dir(self):
+            return getattr(self, key)
+        raise KeyError(key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        """Enable write access through subscription."""
+        if key in self.__dict__:
+            target_type = type(self.__dict__[key])
+            if not isinstance(value, target_type):
+                raise TypeError(
+                    "Mapped value must be subclass of " + target_type.__name__
+                )
```

Comment:
This is interesting. I really like this section. I'm still wondering about some mixes between this and pydantic, but tests should answer those questions. Great!

Reply:
Interesting indeed. Pydantic claims that

> pydantic enforces type hints at runtime

but without my code above, I was able to change DataCacheConfig.hashType (a string) to an int when subscripting. With my code, I got the expected exception. Pydantic didn't react when I changed the value with `DataCacheConfig.hashType =` either. I see that `BaseModel.__setattr__` is decorated with `@no_type_check`, so I added a `__setattr__` with type-checking to `AttrDict`.